src/GDGUkraine/utils.py

```diff
@@ -114,3 +122,283 @@ def make_vcard(user_reg, url=None):
+                                  urllib.parse.quote_plus(str(v))])
+                        for k, v in rkwargs.items() if v])
+
+    if uargs:
+        url = '/'.join([url, uargs])
+
+    if ukwargs:
+        url = '?'.join([url, ukwargs])
+
+    # Return final URI string
+    return url
+
+
+def build_url_map(force=False):
+    '''Builds resolve map for class-based routes
+        build_url_map(force=True) is called by url map builder cherrypy plugin
+    '''
+
+    def retrieve_class_routes(cls, mp, handler_cls=None):
+        if handler_cls is None:
+            handler_cls = '.'.join([cls.__class__.__module__,
+                                    cls.__class__.__name__]).lower()
+        if not mp.endswith('/'):
+            mp = '/'.join([mp, ''])
+        res = {}
+        for method in dir(cls):
+            hndlr = getattr(cls, method)
+            uri = mp
+            if hasattr(hndlr, '__name__') \
+                    and hndlr.__name__ != 'index' \
+                    and method:
+                uri = ''.join([mp, method])
+            if not uri:
+                uri = '/'
+
+            if inspect.ismethod(hndlr):
+                if getattr(hndlr, 'exposed', False):
+                    if hndlr.__name__ != method:
+                        continue
+                    # That's it! It's a final method
+
+                    # (args_, varargs_, varkw_, values_) = \
+                    #     inspect.getargspec(hndlr)
+                    # pprint(inspect.getargspec(hndlr))
+                    # Use inspect.getargspec
+                    # instead of inspect.signature for Python < 3.5
+                    params = inspect.signature(hndlr).parameters
+
+                    key_cls = handler_cls
+                    if hndlr.__name__ != 'index':
+                        key_cls = '.'.join([handler_cls, hndlr.__name__])
+
+                    if res.get(key_cls):
+                        continue
+
+                    res[key_cls] = {
+                        'args': params,
+                        'url': (uri
+                                if uri and uri.endswith('/')
+                                else '/'.join([uri]))}
+            elif not inspect.isfunction(hndlr) and \
+                    not isinstance(hndlr, property) and \
+                    not method.startswith('__'):
+                # Looks like we have another class instance mounted and nested
+                # import ipdb; ipdb.set_trace()
+                res.update(
+                    retrieve_class_routes(
+                        cls=hndlr,
+                        mp=(''
+                            if uri.endswith('/')
+                            else '/').join([uri, method]),
+                        handler_cls='.'.join([handler_cls, method])))
+        # TODO: handle `index` and `default` methods
+        if res.get(handler_cls):
+            res['.'.join([handler_cls, 'index'])] = res[handler_cls]
```

Comment:
why did you do this, instead of removing condition on line 239?

```
if hndlr.__name__ != 'index':
```

Because it seems to me, that here you do EXACTLY THE SAME, you do not do at line 240 due to this condition.